### PR TITLE
Assert not equal status check

### LINKED ISSUE #1317: Assert not equal
Add support for Assert Not Equal for status check. This could be assert_ne. See issue https://github.com/buildtesters/buildtest/issues/1314

### PR DESCRIPTION
**Pull request by prathmesh4321**
Hi @shahzebsiddiqui I have added assert not equal. Please review the PR.
Screen Shot 2023-01-31 at 11 44 22 PM

Screen Shot 2023-01-31 at 11 43 51 PM

### CODE REVIEW

File: buildtest/buildsystem/checks.py [hunk 1]
```diff
@@ -506,6 +506,61 @@ def assert_eq_check(builder):
     return all(assert_check)
 
 
+def assert_ne_check(builder):
+    """This method performs Assert not Equal and is used when ``assert_ne`` property is specified
+    in status check. This method will evaluate each metric value reference value and
+    store assertion in list. The list of assertion is logically AND which will return a True or False
+    for the status check.
+
```
Comment (shahzebsiddiqui) [inline buildtest/buildsystem/checks.py:517]: Please add a `args` section to describe the `builder` argument. you can copy from another method.

File: buildtest/schemas/definitions.schema.json [hunk 2]
```diff
@@ -312,6 +312,36 @@
...
+              "name",
+              "ref"
+            ],
+            "properties": {
+              "name": {
+                "type": "string",
+                "description": "Name of metric to use for comparison"
+              },
+              "ref": {
+                "description": "Specify reference value (str, int,float) for comparison",
```
Comment (shahzebsiddiqui) [inline buildtest/schemas/definitions.schema.json]: add space `int,float` -> `int, float`

File: buildtest/buildsystem/checks.py (hunk 1 above)
Comment (prathmesh4321) [reply buildtest/buildsystem/checks.py:517]: @shahzebsiddiqui Done

File: buildtest/schemas/definitions.schema.json (hunk 2 above)
Comment (prathmesh4321) [reply buildtest/schemas/definitions.schema.json]: Done

**Comment by shahzebsiddiqui** [comment]
can you create a hyperlink for the example test where actual test is in https://github.com/buildtesters/buildtest/tree/devel/tutorials/perf_checks and then the symlink is in https://github.com/buildtesters/buildtest/tree/devel/tests/builders for each test so that the example test is also the one that is tested. 

We can potentially use the one in the `tutorials/perf_checks` as a means for  user-facing docs

**Comment by shahzebsiddiqui** [comment]
@prathmesh4321 please fix the documentation build there is an error on line 319 due to change in filename
Screen Shot 2023-02-01 at 1 54 59 PM

**Comment by prathmesh4321** [comment]
> can you create a hyperlink for the example test where actual test is in https://github.com/buildtesters/buildtest/tree/devel/tutorials/perf_checks and then the symlink is in https://github.com/buildtesters/buildtest/tree/devel/tests/builders for each test so that the example test is also the one that is tested.
> 
> We can potentially use the one in the `tutorials/perf_checks` as a means for user-facing docs

@shahzebsiddiqui I have added symlinks to other tests.

Screen Shot 2023-02-01 at 12 59 15 PM

**Comment by prathmesh4321** [comment]
> @prathmesh4321 please fix the documentation build there is an error on line 319 due to change in filename Screen Shot 2023-02-01 at 1 54 59 PM

Done !

**Review by shahzebsiddiqui — APPROVED** [review]
LGTM

**Pull request merged**